substrate/frame/support/src/traits/tokens/fungible/regular.rs

```diff
@@ -417,18 +422,30 @@ pub trait Balanced<AccountId>: Inspect<AccountId> + Unbalanced<AccountId> {
 	///
 	/// This is just the same as burning and issuing the same amount and has no effect on the
 	/// total issuance.
+	///
+	/// This is infallible, but doesn't guarantee that the entire `amount` is used to create the
+	/// pair, for example in the case where the amounts would cause overflow or underflow in
+	/// [`Balanced::issue`] or [`Balanced::rescind`].
 	fn pair(amount: Self::Balance) -> (Debt<AccountId, Self>, Credit<AccountId, Self>) {
-		(Self::rescind(amount), Self::issue(amount))
+		// Maximum amount which can be issued.
+		let issue_cap = Self::Balance::max_value().saturating_sub(Self::total_issuance());
+		// Maximum amount which can be rescinded.
+		let rescind_cap = Self::total_issuance();
+
+		// Amount which can be both issued and rescinded.
+		let capped_amount = amount.min(issue_cap).min(rescind_cap);
```

Comment:
Hm this is making assumption about when these functions *should* work, but does not *guarantee* it.  
I think its simpler if we return an error (any maybe also take a `Precision`):  
```rust
let issued = Self::issue(amount);
let rescinded = Self::rescind(amount);

if issued.peek() != rescinded.peek() || issued.peek() != amount {
	Err(...)
} else {
	(rescinded, issued)
}
```

It would be possible to also add a `Precision` argument that allows `issued.peek()` to be unequal to `amount`.

Reply:
Thanks for this suggestion, I also prefer this solution.

Reply:
It would still cancel out each other exactly, but the amount could be lower than what was requested.  
Anyway, its probably not *that* important to have.